plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java

```diff
@@ -3467,4 +3471,92 @@ public void testTrinoCacheInvalidatedOnCreateTable()
 
         assertUpdate("DROP TABLE " + tableName);
     }
+
+    @ParameterizedTest
+    @MethodSource("timestampPrecisionOnCreateTableProvider")
+    public void testTimestampPrecisionOnCreateTable(TimestampPrecisionTestSetup setup)
+    {
+        try (TestTable testTable = new TestTable(
+                getQueryRunner()::execute,
+                "test_coercion_show_create_table",
+                "(ts TIMESTAMP)")) {
+            assertUpdate("INSERT INTO " + testTable.getName() + " VALUES (" + setup.sourceValueLiteral() + ")", 1);
+            assertEquals(getColumnType(testTable.getName(), "ts"), setup.newColumnType());
+            assertQuery("SELECT * FROM " + testTable.getName(), "VALUES " + setup.newValueLiteral());
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("timestampPrecisionOnCreateTableProvider")
+    public void testTimestampPrecisionOnCreateTableAsSelect(TimestampPrecisionTestSetup setup)
+    {
+        try (TestTable testTable = new TestTable(
+                getQueryRunner()::execute,
+                "test_coercion_show_create_table_as_select",
+                "AS SELECT %s ts".formatted(setup.sourceValueLiteral()))) {
+            assertEquals(getColumnType(testTable.getName(), "ts"), setup.newColumnType());
+            assertQuery("SELECT * FROM " + testTable.getName(), "VALUES " + setup.newValueLiteral());
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("timestampPrecisionOnCreateTableProvider")
+    public void testTimestampPrecisionOnCreateTableAsSelectWithNoData(TimestampPrecisionTestSetup setup)
+    {
+        try (TestTable testTable = new TestTable(
+                getQueryRunner()::execute,
+                "test_coercion_show_create_table_as_select_with_no_data",
+                "AS SELECT %s ts WITH NO DATA".formatted(setup.sourceValueLiteral()))) {
+            assertEquals(getColumnType(testTable.getName(), "ts"), setup.newColumnType());
+        }
+    }
+
+    public Object[][] timestampPrecisionOnCreateTableProvider()
+    {
+        return timestampPrecisionOnCreateTable().stream()
+                .map(Optional::of)
+                .flatMap(Optional::stream)
+                .collect(toDataProvider());
```

Comment:
We could directly inline them right ? Why converting to `Optional` and use `Optional#stream from it ?

Reply:
Removed the data provided and used loop instead.